tfsdk/attribute_plan_modification_test.go

```diff
@@ -0,0 +1,682 @@
+					"a": tftypes.NewValue(tftypes.String, tftypes.UnknownValue),
+					"b": tftypes.NewValue(tftypes.String, "bar"),
+				}),
+			},
+			config: Config{
+				Schema: schema,
+				Raw: tftypes.NewValue(schema.TerraformType(context.Background()), map[string]tftypes.Value{
+					"a": tftypes.NewValue(tftypes.String, nil),
+					"b": tftypes.NewValue(tftypes.String, "quux"),
+				}),
+			},
+			path:         tftypes.NewAttributePath().WithAttributeName("a"),
+			ifReturn:     true,
+			expectedPlan: types.String{Unknown: true},
+			expectedRR:   false,
+		},
+		"null-config-not-computed": {
+			// if the config is null for a non-computed attribute,
+			// we should require replacing if it's a change.
+			//
+			// unlike computed attributes, this is always a
+			// practitioner making a change, and therefore the
+			// destroy/recreate cycle is likely expected.
+			//
+			// this test is technically covered by
+			// null-attribute-plan, but let's duplicate it just to
+			// be explicit about what each case is actually testing
+			state: State{
+				Schema: schema,
+				Raw: tftypes.NewValue(schema.TerraformType(context.Background()), map[string]tftypes.Value{
+					"a": tftypes.NewValue(tftypes.String, "foo"),
+					"b": tftypes.NewValue(tftypes.String, "bar"),
+				}),
+			},
+			plan: Plan{
+				Schema: schema,
+				Raw: tftypes.NewValue(schema.TerraformType(context.Background()), map[string]tftypes.Value{
+					"a": tftypes.NewValue(tftypes.String, "foo"),
+					"b": tftypes.NewValue(tftypes.String, nil),
+				}),
+			},
+			config: Config{
+				Schema: schema,
+				Raw: tftypes.NewValue(schema.TerraformType(context.Background()), map[string]tftypes.Value{
+					"a": tftypes.NewValue(tftypes.String, "foo"),
+					"b": tftypes.NewValue(tftypes.String, nil),
+				}),
+			},
+			path:         tftypes.NewAttributePath().WithAttributeName("b"),
+			ifReturn:     true,
+			expectedPlan: types.String{Null: true},
+			expectedRR:   true,
+		},
+	}
+
+	for name, tc := range tests {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			attrConfig, diags := tc.config.getAttributeValue(context.Background(), tc.path)
+			if diags.HasError() {
+				t.Fatalf("Got unexpected diagnostics: %s", diags)
+			}
+
+			attrState, diags := tc.state.getAttributeValue(context.Background(), tc.path)
+			if diags.HasError() {
+				t.Fatalf("Got unexpected diagnostics: %s", diags)
+			}
+
+			attrPlan, diags := tc.plan.getAttributeValue(context.Background(), tc.path)
+			if diags.HasError() {
+				t.Fatalf("Got unexpected diagnostics: %s", diags)
+			}
+
+			req := ModifyAttributePlanRequest{
+				AttributePath:   tc.path,
+				Config:          tc.config,
+				State:           tc.state,
+				Plan:            tc.plan,
+				AttributeConfig: attrConfig,
+				AttributeState:  attrState,
+				AttributePlan:   attrPlan,
+				ProviderMeta:    Config{},
+			}
+			resp := &ModifyAttributePlanResponse{
+				AttributePlan: req.AttributePlan,
+			}
```

Comment:
To capture the final conditional, should we augment this to include `RequiresReplace: tc.priorRR` and add a unit testing that has `priorRR` that matches the `tc.path`? Then we should be able to verify prior RR is preserved when `tc.ifReturn` is false.